### PR TITLE
{bio}[GCC/8.2.0-2.31.1] phylokit v1.0, phylonaut v20190626, SVDquest v20190627, ...

### PR DESCRIPTION
**Pull request by omula**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/a/ASTRID/ASTRID-2.2.1-GCC-8.2.0-2.31.1.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+easyblock = 'CMakeMake'
+
+name = 'ASTRID'
+version = "2.2.1"
+
+homepage = 'https://github.com/pranjalv123/ASTRID'
+description = "Improved version of ASTRID"
+
+toolchain = {'name': 'GCC', 'version': '8.2.0-2.31.1'}
```
[the remark this answers is not in the document]
Comment (omula) [reply easybuild/easyconfigs/a/ASTRID/ASTRID-2.2.1-GCC-8.2.0-2.31.1.eb]: @smoors ASTRID has no call to MPI.
Should I always use the "highest" (sub)toolchain of dependencies?

[the remark this answers is not in the document]
Comment (smoors) [reply easybuild/easyconfigs/a/ASTRID/ASTRID-2.2.1-GCC-8.2.0-2.31.1.eb]: @omula yes, that's right.

File: easybuild/easyconfigs/a/ASTRID/ASTRID-2.2.1-GCC-8.2.0-2.31.1.eb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+easyblock = 'CMakeMake'
+
+name = 'ASTRID'
+version = "2.2.1"
+
+homepage = 'https://github.com/pranjalv123/ASTRID'
+description = "Improved version of ASTRID"
```
[the remark this answers is not in the document]
Comment (omula) [reply easybuild/easyconfigs/a/ASTRID/ASTRID-2.2.1-GCC-8.2.0-2.31.1.eb]: changed

**Comment by smoors** [comment]
@omula the test fails because you are using Java 1.8, while other easyconfigs use Java 11 for the same toolchain generation `2019a`.
is it ok to use Java 11, or is version 1.8 really needed?
in the latter case, you have to add a versionsuffix:
```
versionsuffix = '-Java-%(javaver)s'
```

**Comment by omula** [comment]
@smoors as far as I can see the package does not need Java but is CMake checking for it.
```CMake Error at /beegfs/easybuild/CentOS/7.6.1810/Skylake/software/CMake/3.13.3-GCCcore-8.2.0/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find Java (missing: Java_JAVA_EXECUTABLE Java_JAR_EXECUTABLE
  Java_JAVAC_EXECUTABLE Java_JAVAH_EXECUTABLE Java_JAVADOC_EXECUTABLE)
Call Stack (most recent call first):
  /beegfs/easybuild/CentOS/7.6.1810/Skylake/software/CMake/3.13.3-GCCcore-8.2.0/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /beegfs/easybuild/CentOS/7.6.1810/Skylake/software/CMake/3.13.3-GCCcore-8.2.0/share/cmake-3.13/Modules/FindJava.cmake:331 (find_package_handle_standard_args)
```

Should I disable this check? Is there any way to disable this check?

**Comment by smoors** [comment]
@omula from the release notes:
```
The -nojava version does not allow for BIONJ* to be used.
```
so, some functionality clearly needs it (see phydstar.cpp), although I don't know how important it is.
you could patch the CMakeLists.txt file to prevent it from building the Java related code,
but why not just try to build it with Java 11?

**Comment by omula** [comment]
@smoors thanks for pointing it out. I don't know how did I make the search before so that I couldn't see anything Java related in the code. I did the search again and I did see them.

Readded Java dependency with version 11.

File: easybuild/easyconfigs/a/ASTRID/ASTRID-2.2.1-gompi-2019a-Java-11.eb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+easyblock = 'CMakeMake'
+
+name = 'ASTRID'
+version = "2.2.1"
+versionsuffix = '-Java-%(javaver)s'
```
Comment (smoors) [inline easybuild/easyconfigs/a/ASTRID/ASTRID-2.2.1-gompi-2019a-Java-11.eb]: @omula I'm sorry if I was not clear enough.
the versionsuffix is only used if the Java version differs from the default for a given toolchain generation. for `2019a` the default is Java 11, so you should not add the versionsuffix.

Comment (omula) [reply easybuild/easyconfigs/a/ASTRID/ASTRID-2.2.1-gompi-2019a-Java-11.eb]: @smoors changed!
Btw, how do I know which is the default Java version given a toolchain? Is there also a default Python version?
Thanks

Comment (smoors) [reply easybuild/easyconfigs/a/ASTRID/ASTRID-2.2.1-gompi-2019a-Java-11.eb]: @omula thanks!

I don't know if there is an official list. I just grep the easyconfigs in the develop branch.
the first version of any software that is used as dependency is considered the default. other versions are not allowed, with exceptions for ['ASE', 'Boost', 'Java', 'Lua', 'PLUMED', 'R'].
so yes, there is also a default Python version :)

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node302.hydra.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/77ecb2ebc0db0b89df462261383c170b for a full test report.

**Pull request merged**